Add doc for upgrade-assistant telemetry (#24758)
dotnet/docs: docs/core/porting/upgrade-assistant-telemetry.md

---
title: Upgrade Assistant Telemetry
description: Learn about telemetry collected by the Upgrade Assistant.
author: tasou
ms.date: 06/03/2021
---
# Upgrade Assistant telemetry

> The [Upgrade Assistant](xref:upgrade-assistant-overview) includes a telemetry feature that collects usage data. It's important that the Upgrade Assistant team understands how the tool is used so it can be improved.

Reviewer: ```suggestion
The [Upgrade Assistant](xref:upgrade-assistant-overview) includes a telemetry feature that collects usage data. The telemetry data is used to help understand how to make improvements to the tool.
```

| >=0.2.231802               | Kernel version.                                                                                             |
| >=0.2.231802               | Upgrade Assistant version.                                                                                  |
| >=0.2.231802               | The command and argument names invoked. Actual argument values aren't collected.                            |
| >=0.2.231802               | MSBuild version used.                                                                                       |
| >=0.2.231802               | Hashed solution id (or hashed path if no id is available).                                                  |
| >=0.2.231802               | Hashed project id (or hashed path if no id is available) for each project.                                  |
| >=0.2.231802               | Hashed project id (or hashed path if no id is available) for each entrypoint.                               |
| >=0.2.231802               | For each step, the time to initialize and apply the step.                                                   |

> | >=0.2.231802               | For each step, the the decision selected (for example, `apply`).                                            |

Reviewer: ```suggestion
| >=0.2.231802               | For each step, the decision selected (for example, `apply`).                                            |
```
